internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -191,3 +181,52 @@ func (c *CodePipeline) getPipelineName(resourceArn string) (string, error) {
 
 	return parsedArn.Resource, nil
 }
+
+// GetPipelineStatus retrieves status information from a given pipeline.
+func (c *CodePipeline) GetPipelineState(name string) (*PipelineState, error) {
+	input := &cp.GetPipelineStateInput{
+		Name: aws.String(name),
+	}
+	resp, err := c.client.GetPipelineState(input)
+
+	if err != nil {
+		return nil, fmt.Errorf("get pipeline state %s: %w", name, err)
+	}
+	var stageStates []*StageState
+	for _, stage := range resp.StageStates {
+		var bool string
```

Comment:
i think it's better to avoid using reserved name like `bool`, `int` as var names.

Reply:
Ah, makes sense. Thanks!